### PR TITLE
kbpages: add KVStore backed cert store

### PR DESCRIPTION
**Pull request by songgao**
This store (`autocert.Cache`) is used both as an actual cert storage, and for storing temporary tokens for cert provisioning. So this would allow us to move toward being able to run multiple instances.

This doesn't use it yet. I'll have a separate PR to move away from CLI arg config.

### CODE REVIEW

**Comment by jzila** [comment]
I didn't review the code but why are you adding a KVStoreClient directly to `libkbfs`? Why not only have it in `libpages`?

**Comment by songgao** [comment]
> I didn't review the code but why are you adding a KVStoreClient directly to `libkbfs`? Why not only have it in `libpages`?

`libpages` uses `libkbfs.Config` and this seems to be the only way to get it without creating a new RPC connection. It's the same RPC connection anyway, just a new client struct. Could've added a flag to skip making the KVStore client but I didn't wanna make init'ing libkbfs more complicated for this.

**Comment by jzila** [comment]
Got it, thanks

File: go/kbfs/libpages/cert_store_backed_by_kvstore.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	"encoding/base64"
+
+	"github.com/keybase/client/go/kbfs/libkbfs"
+	"github.com/keybase/client/go/protocol/keybase1"
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/acme/autocert"
+)
+
+// certStoreBackedByKVStore is a wrapper around the KVStore in Keybase. No
+// caching is done here since acme/autocert has a in-memory cache already.
```
Comment (strib) [inline go/kbfs/libpages/cert_store_backed_by_kvstore.go:18]: "a in-memory" -> "an in-memory"

File: go/kbfs/libpages/cert_store_backed_by_kvstore.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+type certStoreBackedByKVStore struct {
+	kbfsConfig libkbfs.Config
```
Comment (strib) [inline go/kbfs/libpages/cert_store_backed_by_kvstore.go:20]: Maybe just pass in and store the kvstoreclient here?  Would be better for testing, if we ever add tests for it.

**Comment by songgao** [comment]
> Cool, looks good mod two nits. Making sure I understand: there's no way to actually configure the server to use this cache yet, right? The main function seems to only look at a boolean still.

That's correct. I'm planning on changing it to using a json config file instead of CLI, so it's easier to deploy. So didn't end up enabling that for the binary yet.

File: go/kbfs/libpages/cert_store_backed_by_kvstore.go (hunk 2 above)
Comment (songgao) [reply go/kbfs/libpages/cert_store_backed_by_kvstore.go:20]: I was a bit worried that something along `s.kbfsConfig.KeybaseService().GetKVStoreClient()` might be switched out when the process is running, but after a few greps it looks like they never change once init'ed, so it's probably OK I guess. Could you confirm?

Comment (songgao) [reply go/kbfs/libpages/cert_store_backed_by_kvstore.go:20]: Looking at this more, it seems `kbpki_client.go` always calls `KeybaseService()`: https://github.com/keybase/client/blob/ea7eea5070a2dd74a75694352c2f385bfa0945ba/go/kbfs/libkbfs/kbpki_client.go#L59 so perhaps this should too? I guess we could define a service owner here too like in kbpki.

**Pull request merged**